### PR TITLE
Added disability_transformation

### PR DESCRIPTION
**Pull request by raft001**
This transformation changes outdated names of disabilities with more appropriate ones.

### CODE REVIEW

File: transformations/disability_transformation/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Disability/Differently Abled Transformation ♿
+This transformation changes outdated names of disabilities with more appropriate ones which are now used.
+
```
Comment (kaustubhdhole) [inline transformations/disability_transformation/README.md:3]: Please add your name/email/affiliation.

Comment (raft001) [reply transformations/disability_transformation/README.md:3]: Added it. 
Thanks

File: transformations/disability_transformation/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+## What type of a transformation is this?
+This transformation replaces outdated names for disabilities with more considerate names now used in the modern world.
+For e.g. 
+If the input sentence is: "After the accident, she is now a cripple."
+The output will be: "After the accident, she is now a person with physical disability.".
+
+
+## What tasks does it intend to benefit?
+This transformation would benefit all tasks which intend to generate datasets with more respectful versions of terms involving people with disabilities, resulting in more diverse datasets.
```
Comment (samsontmr) [inline transformations/disability_transformation/README.md:15]: I would say "less biased/offensive" rather than "more diverse".

File: transformations/disability_transformation/test.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        "sentence": "He is so slow he's probably a person with a learning disablity "
+      }]
+    },
+    {
+      "class": "DifferentAbilityTransformation",
+      "inputs": {
+        "sentence": "John is deaf."
+      },
+      "outputs": [{
+        "sentence": "John is  person with a hearing disability "
```
Comment (samsontmr) [inline transformations/disability_transformation/test.json:28]: You might need to add some functionality to do cleanup: there is an extra space before and after the replaced phrase and the punctuation was removed.

File: transformations/disability_transformation/transformation.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+                           "wacko":" person with a mental impairment ",
+                           "nuts":" person with a mental impairment ",
+                           "retard":" person with an intellectual  disablity ",
+                           "retards":" people with an intellectual  disablity ",
+                           "retarded":" person with an intellectual  disablity "
+                           }
+!python -m spacy download en_core_web_sm
+# POS TAGGER
+import spacy
+from spacy.lang.en.examples import sentences
```
Comment (samsontmr) [inline transformations/disability_transformation/transformation.py:51]: Move import statements to the top of the file.

File: transformations/disability_transformation/transformation.py [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+                           "retards":" people with an intellectual  disablity ",
+                           "retarded":" person with an intellectual  disablity "
+                           }
+!python -m spacy download en_core_web_sm
+# POS TAGGER
+import spacy
+from spacy.lang.en.examples import sentences 
+nlp = spacy.load("en_core_web_sm")
+def postag(text):
+  doc = nlp(text)
```
Comment (samsontmr) [inline transformations/disability_transformation/transformation.py:54]: Can we please use 4 spaces instead of 2?

File: transformations/disability_transformation/transformation.py [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+  pos_list=[]
+  word_list=[]
```
Comment (samsontmr) [inline transformations/disability_transformation/transformation.py:56]: There should be a space between operators and after commas

**Review by samsontmr — CHANGES_REQUESTED** [review]
Thanks for the contribution @raft001! I've left some comments.

**Comment by raft001** [comment]
Thanks for the review  @samsontmr It was very helpful and I have made the required changes.
I am looking forward to any other suggestions.

File: transformations/disability_transformation/transformation.py [hunk 7]
```diff
@@ -0,0 +1,213 @@
+!python -m spacy download en_core_web_sm
+
+import spacy
+import string
+from spacy.lang.en.examples import sentences 
+nlp = spacy.load("en_core_web_sm")
+
+
```
Comment (samsontmr) [inline transformations/disability_transformation/transformation.py:8]: Reduce the number of new lines

File: transformations/disability_transformation/transformation.py [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+
+
+# A dict containing offensive words and their alternatives
+disability_names =  {"blind":"person or people with a visual impairment",
```
Comment (samsontmr) [inline transformations/disability_transformation/transformation.py:12]: I wonder if it would be a good idea to move this dict into an external json file so it can be edited separately? What do you think @kaustubhdhole?

**Review by samsontmr — COMMENTED** [review]
Another general style comment: there should be spaces after commas.

File: transformations/disability_transformation/transformation.py (hunk 8 above)
Comment (kaustubhdhole) [reply transformations/disability_transformation/transformation.py:12]: The dictionary is small anyways so would be fine both ways.

File: transformations/disability_transformation/transformation.py [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+        self.disability_names = disability_names
+
+    def generate(self, sentence: str):
+
+      return [different_ability(sentence, self.disability_names)]
+    
+  
+  
+  
+if __name__ == '__main__':
```
Comment (kaustubhdhole) [inline transformations/disability_transformation/transformation.py:191]: This can be commented or removed.

Comment (raft001) [reply transformations/disability_transformation/transformation.py]: I have made the following changes: removed excessive new lines, added spaces after commas and removed this end part.
 Thanks.

File: transformations/disability_transformation/transformation.py [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION
+    ]
+    languages = ["en"]
+
+    def __init__(self, seed=0, max_outputs=1):
+        super().__init__(seed, max_outputs = max_outputs)
+        self.disability_names = disability_names
+        self.nlp = spacy.load("en_core_web_sm")
```
Comment (kaustubhdhole) [inline transformations/disability_transformation/transformation.py:141]: Please load spacy like [this](https://github.com/GEM-benchmark/NL-Augmenter/blob/bb4dc6c5000aad54ded232549f3dee4ff3d2b972/transformations/hashtagify/transformation.py#L114)

File: transformations/disability_transformation/transformation.py [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+            wl, pl = postag(text)
+            max_len = len(wl)
+            indices = get_index(wl, name)
+            text = placement(indices, wl, pl, input, disability_names, name)         
+        text = preserve_capitalization(input, text)
+        text = restore_punc(text)
+    return text
+
+class DifferentAbilityTransformation(SentenceOperation):
+    tasks = [
```
Comment (kaustubhdhole) [inline transformations/disability_transformation/transformation.py:132]: Please add the revelant [keywords](https://github.com/GEM-benchmark/NL-Augmenter/blob/main/docs/keywords.md) for your PR too.

File: transformations/disability_transformation/transformation.py [hunk 12]
```diff
@@ -0,0 +1,144 @@
+!python -m spacy download en_core_web_sm
+from interfaces.SentenceOperation import SentenceOperation
+from evaluation.evaluation_engine import evaluate, execute_model
+from tasks.TaskTypes import TaskType
+import spacy
+import string
+from spacy.lang.en.examples import sentences 
+nlp = spacy.load("en_core_web_sm")
```
Comment (kaustubhdhole) [inline transformations/disability_transformation/transformation.py:8]: remove this spacy loading

File: transformations/disability_transformation/transformation.py [hunk 13]
```diff
@@ -0,0 +1,144 @@
+!python -m spacy download en_core_web_sm
```
Comment (kaustubhdhole) [inline transformations/disability_transformation/transformation.py:1]: You don't need to add the first line.

**Comment by kaustubhdhole** [comment]
Overall, this is an important PR! Thanks for your hard work @raft001. Please pull main into your branch and address the above suggestions and I am happy to merge.

File: transformations/disability_transformation/transformation.py [hunk 14]
```diff
@@ -1,173 +1,133 @@
...
+    for i in string.punctuation:
+        text = text.replace(" "+i,i)
+    return text
+
 
 def different_ability(input, disability_names):
-  #lowercase
     text = input.lower()
-
+    text=replace_punc(text)
```
Comment (samsontmr) [inline transformations/disability_transformation/transformation.py:120]: Could you add spaces around the `=`?

Comment (raft001) [reply transformations/disability_transformation/transformation.py]: Added them!

**Comment by raft001** [comment]
> Overall, this is an important PR! Thanks for your hard work @raft001. Please pull main into your branch and address the above suggestions and I am happy to merge.

I've made the required changes, and the build is successful now. Thanks

**Review by kaustubhdhole — APPROVED** [review]

File: transformations/disability_transformation/transformation.py [hunk 15]
```diff
@@ -0,0 +1,151 @@
+from interfaces.SentenceOperation import SentenceOperation
+from evaluation.evaluation_engine import evaluate, execute_model
+from tasks.TaskTypes import TaskType
+import spacy
+import string
+from spacy.lang.en.examples import sentences 
+
+# A dict containing offensive words and their alternatives
+disability_names =  {"blind":"person or people with a visual impairment", 
+                           "deformed": "person or people with a physical disability",
```
Comment (kaustubhdhole) [inline transformations/disability_transformation/transformation.py:10]: One suggestion to make your PR stronger, I think in the cases in which you can determine the tense of the subject, you can probably choose just one between "person.." or "people.."? If you are interested you can do that in a separate PR..
